src/trees/NodeAllocator.cpp

```diff
@@ -130,7 +130,7 @@ template <int D, typename T> int NodeAllocator<D, T>::alloc(int nNodes, bool coe
 
     // we require that the index for first child is a multiple of 2**D
     // so that we can find the sibling rank using rank=sIdx%(2**D)
-    if (sIdx % nNodes != 0) MSG_ERROR(" node allocate error");
+    if (sIdx % nNodes != 0) std::cout << "Warning: recommended number of siblings is 2**D" << std::endl;
```

Comment:
I guess we should use the `MSG_` macros that Stig made :)
```suggestion
    if (sIdx % nNodes != 0) MSG_WARN("Warning: recommended number of siblings is 2**D");
```